src/parser/SparqlParser.cpp

```diff
@@ -507,6 +507,31 @@ void SparqlParser::addFilter(const string& str, vector<SparqlFilter>* _filters,
         f._type = SparqlFilter::REGEX;
         f._lhs = lhs;
         f._rhs = rhs.substr(1, rhs.size() - 2);
+        if (f._rhs[0] == '^') {
+          // Check if we can use the more efficient prefix filter instead of
+          // an expensive regex filter.
+          bool isSimple = true;
+          bool escaped = false;
+          for (size_t i = 1; isSimple && i < f._rhs.size(); i++) {
+            if (f._rhs[i] == '\\') {
+              escaped = true;
+              continue;
+            }
+            if (!escaped) {
+              char c = f._rhs[i];
+              std::string badChars = "[^$.|?*+()";
```

Comment:
I'm not sure if 10 chars always fit inside the small string opt. maybe this should be a `constexpr std::string` so we definitely don't alloc in a loop here. Then it might also make sense to instead use a `std::array<bool, std::numeric_limits<unsigned char>::max()>` and looking up the `static_cast<unsigned char>(c)` index. Then again all of that is probably premature optimization since this is only the `SparqlParser` and not some tight filer loop